### PR TITLE
19 prevent vertical scrolling

### LINKED ISSUE #19: Prevent vertical scrolling of main viewport when a dialog is open
When a dialog is open we should append a state hook e.g. `is-dialog-open` to the `<html>` element which allows for the following CSS to be applied:

``` css
html.is-dialog-open {
  overflow-x: hidden;
}
```

This prevents the main viewport being scrolled up and down when a dialog is open which is standard UX behaviour.

### PR DESCRIPTION
**Pull request by DaveOrDead**
Fixes #19 

Switched casing for constant
Prevent scroll when dialog open
prevent scroll

### CODE REVIEW

File: demo/css/main.css [hunk 1]
```diff
@@ -34,3 +34,7 @@ body {
 .dialog.is-active {
     display: block !important;
 }
+
+.is-no-scroll {
```
Comment (chris-pearce) [inline demo/css/main.css]: I reckon including `vertical` in this class has value and is `has` better than `is` e.g. `.has-no-vertical-scroll`? Plus we'll need to chain it to the element it's being applied too which should be the `<html>` element. 

It actually makes me wonder if we'll need to namespace our state hooks also as not all state hooks will be chained to a namespaced style hook, like in this instance.

``` css
html.has-no-vertical-scroll
```

I'll write up a **discussion/question** type issue for this.

Comment (DaveOrDead) [reply demo/css/main.css]: maybe it should be a helper instead:

`h-prevent-vertical-scroll`

Comment (chris-pearce) [inline demo/css/main.css]: FYI: https://github.com/fedordead/vanilla-ui/issues/32

File: src/components/dialog/dialog.js [hunk 2]
```diff
@@ -138,6 +138,9 @@ const VUIDialog = ({
         // Add class to make dialog visible. Needs to occur before focus.
         dialog.classList.add(activeClass);
 
+        // Prevent page scroll
```
Comment (chris-pearce) [inline src/components/dialog/dialog.js]: Nitpicking but really not a fan of "page", I'd go with "viewport" and use "Remove" as that seems to be the more consistent word being used throughout, but that's just me 😄.

File: src/components/dialog/dialog.js [hunk 3]
```diff
@@ -138,6 +138,9 @@ const VUIDialog = ({
...
+        DOM.page.classList.add(DEFAULT_CLASSNAMES.isNoScroll);
```
Comment (chris-pearce) [inline src/components/dialog/dialog.js]: Already discussed `DOM.page` naming 👍

File: demo/css/main.css (hunk 1 above)
Comment (chris-pearce) [reply demo/css/main.css]: Not sure if it makes sense to introduce the concept of helpers in this library's CSS and I don't think there's gonna be any reuse of this anyway. I think a state hook makes sense for this.

**Comment by chris-pearce** [comment]
LGTM

**Pull request merged**